src/main/java/org/openstreetmap/atlas/checks/validation/tag/RoadNameSpellingConsistencyCheckWalker.java

```diff
@@ -111,6 +120,7 @@ public int getValue()
                 .distanceTo(startEdge.start().getLocation())
                 .isLessThanOrEqualTo(maximumSearchDistance)
                         ? incomingEdge.connectedEdges().stream().filter(Edge::isMasterEdge)
+                                .filter(e -> !exploredEdges.contains(e.getMasterEdgeIdentifier()))
```

Comment:
nit: Instead of filtering by master edge and then filtering by exploredEdges, couldn't we combine the two filter?```incomingEdge.connectedEdges().stream().filter(connectedEdge ->connectedEdge.isMasterEdge() &&  !exploredEdges.contains(connectedEdge.getMasterEdgeIdentifier()))```